### ISSUE TITLE
Broken replays at replays.wesnoth.org (GNA #19603)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1300]
**[Original submission](https://gna.org/bugs/index.php?19603)** by slowthinker on 2012-03-30

A replay gets broken if the hosting player disconnected.

How to reproduce:

1. start a MP game on the official server

2. disconnect the hosting player (for example kill wesnothd.exe) 

3. end the game

4. check the replay at replays.wesnoth.org

More details here:

http://forums.wesnoth.org/viewtopic.php?f#4&t36499

(Reproduced on Win XP Pro)
Release: 1.10.1
Priority: 5 - Normal
Severity: 4 - Important

**Comment by wesnoth-bugs** [comment]
**Modified on 2012-04-01**

**slowthinker wrote:**

> It looks my original submission was wrong about the way how to reproduce, and the fact whether the hosting player disconnected or not has nothing to do with the bug:
> 
> The bug happens if the game has been reloaded.
> 
> 
> 
> Summary of my current understanding about the bug:
> 
> http://forums.wesnoth.org/viewtopic.php?p=525335#p525335
> 
> 
> 
> ----------------------------------------------------------
> 
> The bug has a common point with this bug: https://gna.org/bugs/?19258

**Comment by wesnoth-bugs** [comment]
**Modified on 2014-03-12**

**slowthinker wrote:**

> tested again with 1.10.7 (but I guess the bug is not related to a Wesnoth version)
> 
> 
> 
> this game has been reloaded:
> 
> http://replays.wesnoth.org/1.10/20140312/2p__Den_of_Onis_Turn_2_(10571).gz
> 
> 
> 
> the final and corrupted replay:
> 
> http://replays.wesnoth.org/1.10/20140312/2p__Den_of_Onis_Turn_3_(10573).gz

**Comment by wesnoth-bugs** [comment]
**Modified on 2014-04-09**

**involution changed severity:** 3 - Normal -> 4 - Important

**involution changed status:** None -> In Progress

**Comment by wesnoth-bugs** [comment]
**Modified on 2014-04-27**

**involution changed status:** In Progress -> Postponed

**involution wrote:**

> Status:
> 
> 
> 
> This has been partially fixed, in that "basic" scenarios should now work in 1.12. We *may* be able to easily fix campaigns. Currently only the final scenario of the campaign will be saved.
> 
> 
> 
> tl,dr:
> 
> 
> 
> Making reloaded scenarios work would unfortunately take a good deal of work, and this is something we would have to postpone until after 1.12. The good news is that we can patch the server to fix things like this without breaking compatibility, so it doesn't have to happen before 1.12 is launched.
> 
> 
> 
> Long version:
> 
> 
> 
> The issue is basically that the replay server is not smart enough to read the replay and keep track of the game state. The replay that is generated is closer to some kind of debugging output -- the [replay] contains all of the "history" of game transactions, and the [replay_start] contains the initial level cfg generated by the host.
> 
> 
> 
> This is compounded by the fact that the "history" needs to be okay to send to observers in order to bring them up to date. It's possible that the server could generate replay files by always viewing turn 1 as the start, and merging the "new history" with the reloaded history, but that surgery would have to be done in the final seconds before writing the replay to memory, and it may make it harder to debug other server issues before 1.12.0, so I think this kind of change should wait until after that.

**Comment by wesnoth-bugs** [comment]
**Modified on 2014-05-05**

**slowthinker wrote:**

> The next text is partially off topic, and a feature request.
> 
> 
> 
> Think about situations where a player/observer replays
> 
> a) from the very beginning (that happens in an off-line replay, from a saved game)
> 
> b) from the last load point (that happens in an on-line replay, from server's data)
> 
> 
> 
> A replay from the very beginning might be useful also for an on-line observer (in fact from another Wesnoth player I got a question how to do it).
> 
> A replay from the last load point might be useful also for an offline replay: it would allow PBEM games ...
> 
> 
> 
> (... but only if one could watch the replay from a "Point of view" of side 2, 3 ,4 ... I don't understand why there is the limitation Team 1/Each Team/Full Map;
> 
> the choice of a viewpoint would be useful also online - a player who disconnected could rejoin safely and without the blindfold mode, and in the meantime the other player could continue his turn.)

**Comment by wesnoth-bugs** [comment]
**Modified on 2015-01-07**

**gfgtdf wrote:**

> the original bug is most likeley already fixed in master (1.13-dev)

**Comment by wesnoth-bugs** [comment]
**Modified on 2016-03-04**

**ben24626 wrote:**

> I posted the same issue to the wesnoth forums (http://forums.wesnoth.org/viewtopic.php?f#4&t43871) and was advised to bump this post. Also I tried the dev version and had no luck.

**Comment by wesnoth-bugs** [comment]
**Modified on 2016-03-04**

**gfgtdf wrote:**

> Yes there are multiple bugs related to downloaded replays. Most (all) of these are fixed in 1.13. But this doesnt mean that 1.13 can read broken replays generated by wesnoth 1.12.x
> 
> The list of known issues includes:
> 
> 1) replays of games that use modifications or eras with events miss those events causing OOS
> 
> 2) replays of reloaded games are corrupt.
> 
> 
> 
> In both of the cases above it is afaik possible to 'repair' those replay files by hand using a texteditor.
> 
> 
> 
> I don't think that the 1.12 server will ever be fixed to generate valid replays in those cases. What we could try is to add a compability path for those files in the loading savefile code. But again i dont think thi is likeley to happen
